### DISCUSSION TITLE
Building my WPF Pivot Grid in Avalonia

### DISCUSSION THREAD

**Discussion by leaderanalytics** [AvaloniaUI/Avalonia#8097] [General]
Longtime WPF developer, just getting started with Avalonia.  Very sorry for asking these basic questions but I have to get started somewhere.  

I am porting my [WPF pivot grid control](https://github.com/leaderanalytics/LeaderPivot.XAML.WPF) to [Avalonia](https://github.com/leaderanalytics/LeaderPivot.XAML.Avalonia).

WPF has the concept of WPF Control Library project type.  I find no analogue in Avalonia.  So I am using a [net6 library project](https://github.com/leaderanalytics/LeaderPivot.XAML.Avalonia/blob/main/LeaderPivot.XAML.Avalonia/LeaderPivot.XAML.Avalonia.csproj).  I just copied dependencies from another project.  Is this correct?

[My control](https://github.com/leaderanalytics/LeaderPivot.XAML.Avalonia/blob/main/LeaderPivot.XAML.Avalonia/LeaderPivotControl.axaml) is just Hello World at this point but I can't get it to render in [MainWindow of my app](https://github.com/leaderanalytics/LeaderPivot.XAML.Avalonia/blob/main/LeaderPivot.XAML.Avalonia.Host/Views/MainWindow.axaml).  What am I missing here?

Last thing:  I feel very fortunate to have found [this article](https://www.codeproject.com/Articles/5330464/Running-and-Debugging-Multiplatform-NET-NET-Core-N) showing how to publish/build/run my app on WSL.  Hopefully there will be more documentation like this in the near future.  

[Reference](https://github.com/AvaloniaUI/Avalonia/issues/2516)

**Comment by kekekeks** [comment]
1) This should be sufficient for a control library:
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <Nullable>enable</Nullable>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Avalonia" Version="0.10.13" />
  </ItemGroup
</Project>
```
2) You need to manually include your styles from App.xaml using StyleInclude. Styles are not picked up automatically.

**Comment by leaderanalytics** [comment]
Thanks @kekekeks, however  I am still missing something as both the below do not work:


	<Application xmlns="https://github.com/avaloniaui"
				 xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
				 xmlns:local="using:LeaderPivot.XAML.Avalonia"
				 x:Class="LeaderPivot.XAML.Avalonia.App">
		<Application.Styles>
			<StyleInclude Source="local://LeaderPivot.XAML.Avalonia/LeaderPivotControl.axaml" />
		</Application.Styles>
	</Application>





	<Application xmlns="https://github.com/avaloniaui"
				 xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
				 xmlns:local="using:LeaderPivot.XAML.Avalonia"
				 x:Class="LeaderPivot.XAML.Avalonia.App">
		<Application.Styles>
			<StyleInclude Source="avares://LeaderPivot.XAML.Avalonia/LeaderPivotControl.axaml" />
		</Application.Styles>
	</Application>
